src/Illuminate/Foundation/Http/Middleware/TransformsRequest.php

```diff
@@ -63,11 +63,16 @@ protected function cleanParameterBag(ParameterBag $bag)
      * Clean the data in the given array.
      *
      * @param  array  $data
+     * @param  string  $parentKey
      * @return array
      */
-    protected function cleanArray(array $data)
+    protected function cleanArray(array $data, $parentKey = '')
     {
-        return collect($data)->map(function ($value, $key) {
+        return collect($data)->map(function ($value, $key) use ($parentKey) {
+            if (is_int($key) && $parentKey != '') {
```

Comment:
```php
0 != ''
// false
```

So the numeric array behavior will end up being inconsistent:

```php
$key = 3;
$parentKey = 2;
is_int($key) && $parentKey != '';
// true
```

```php
$key = 3;
$parentKey = 0;
is_int($key) && $parentKey != '';
// false
```

Since `$except` is for readable named request parameters, would make more sense to explicitly look for a string `$parentKey` value from an associative array?

```php
protected function cleanArray(array $data, $parentKey = null)
{
    return collect($data)->map(function ($value, $key) use ($parentKey) {
        $key = is_int($key) && is_string($parentKey) ? $parentKey : $value;

        return $this->cleanValue($key, $value);
    })->all();
}
```

Reply:
Yes 😅